x-pack/legacy/plugins/maps/public/connected_components/layer_panel/join_editor/resources/join_expression.js

```diff
@@ -134,10 +133,6 @@ export class JoinExpression extends Component {
       return null;
     }
 
-    const filterStringOrNumberFields = field => {
```

Comment:
Removing this restriction on right join fields. Should be able to perform a join for any value that can be used in a terms aggregation